src/ast.rs

```diff
@@ -368,6 +368,12 @@ pub enum ClauseGuard<Type> {
         typ: Type,
         name: String,
     },
+
+    IntLiteral {
```

Comment:
Tiny nitpick, could we call this `Int`? Thanks

Reply:
Done and done!